### PR TITLE
(React) dynamic-list-of-goods

### PR DESCRIPTION
**Pull request by Ostafii4uk**
[DEMO LINK](https://Ostafii4uk.github.io/react_dynamic-list-of-goods/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,12 +1,58 @@
...
+      });
+  };
+
+  const get5FirstUsers = () => {
+    get5First()
+      .then(user => {
+        setGoods(user.sort(
+          (userA, userB) => userA.name.localeCompare(userB.name),
+        ).slice(0, 5));
+      });
```
[the remark this answers is not in the document]
Comment (Ostafii4uk) [reply src/App.tsx]: Ok, thank u

File: src/App.tsx [hunk 2]
```diff
@@ -1,12 +1,48 @@
...
-);
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-export default App;
+  const loadUser = (callback:() => Promise<Good[]>) => {
+    callback()
+      .then(user => {
+        setGoods(user);
+      });
```
Comment (solaryasha) [inline src/App.tsx:14]: user ?

File: src/App.tsx [hunk 3]
```diff
@@ -1,12 +1,48 @@
...
+  };
```
Comment (solaryasha) [inline src/App.tsx:15]: Do we load user or something different ?

**Review by solaryasha — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 3 above)
Comment (Ostafii4uk) [reply src/App.tsx]: Something different 😄

**Review by solaryasha — APPROVED** [review]

File: src/App.tsx [hunk 4]
```diff
@@ -1,12 +1,48 @@
...
+import { GoodsList } from './components/GoodsList';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-export default App;
+  const loadUser = (callback:() => Promise<Good[]>) => {
```
[the remark this answers is not in the document]
Comment (Ostafii4uk) [reply src/App.tsx:10]: Exactly 👍

**Pull request closed without merging**